wordpress-mobile/WordPress-iOS
# Add domain credit section & UI to be shown on site dashboard

Refs. #11467 - subtask - add new subsection UI

* Changes in `BlogDetailsSection`:
  * Created `BlogDetailsSectionCategory` enum to identify each table view section in `BlogDetailsSection` view model.
  * Added optional `footerTitle`, default to `nil` (no footer title).
  * Also updated `title` to `nullable`, some existing sections have no title and pass in `nil`.
* Changes in `BlogDetailsRow`:
  * Added `imageColor` and `hasNoAccessory` (explicitly specifying no accessory) with default values to keep existing style.
  * Added initializer for domain credit usage in Swift extension.
* Created `BlogDetailsViewController+DomainCredit` extension for domain credit section view model and a placeholder action handler.
* Created `BlogDetailsViewController+SectionHelpers` extension to help map subsection/section enum to table view section index:
  * Mapping from `BlogDetailsSubsection` to `BlogDetailsSectionCategory` with unit tests to ensure mapping for all subsections.
  * Finding index of section view model that matches the target section category enum with unit tests.
* Changes in `BlogDetailsViewController`:
  * Showed domain credit section under feature flag.
  * Used section index helper to get table view section index from visible section view models.

To test:
- Domain credit UI steps:
  - Go to a site/blog dashboard
  - When `domainCredit` feature flag is off (off on this branch), should not see any domain credit UI
  - Turn on `domainCredit` feature flag in `FeatureFlag` enum, should see domain credit section below the site/blog name
  - If QuickStart feature is off, manually turn it on in `BlogDetailsViewController`'s `shouldShowQuickStartChecklist` function. Domain credit section should appear above QuickStart section
- This PR also has changes in the following scenarios, and might be worth testing:
  - When `Blog` model changes so that QuickStart's visibility changes (under these conditions [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift#L33))
  - `restorableSelectedIndexPath` in case of app state restoration (would like to know how to test this)
  - `SearchManager` scenarios linking to a blog that shows certain subsection via [Spotlight search](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/System/WordPressAppDelegate.m#L251) - might require a device

Update release notes:


Example screenshot:

![Simulator Screen Shot - iPhone SE - 2019-04-25 at 10 33 12](https://user-images.githubusercontent.com/1945542/56733078-b40a3f00-6791-11e9-919a-513dc1b26eb7.png)
